server/api.go

```diff
@@ -493,6 +503,114 @@ func (p *Plugin) getYourPrs(w http.ResponseWriter, r *http.Request) {
 	w.Write(resp)
 }
 
+func (p *Plugin) getYourPrsExtraInfo(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	ctx := context.Background()
+
+	var githubClient *github.Client
+
+	if info, err := p.getGitHubUserInfo(userID); err != nil {
+		writeAPIError(w, err)
+		return
+	} else {
+		githubClient = p.githubConnect(*info.Token)
+	}
+
+	var prInfo []*PRExtraInfo
+	json.NewDecoder(r.Body).Decode(&prInfo)
```

Comment:
Why is this `PRExtraInfo` struct? Is this not the list of yourPRs that are passed in?

Reply:
The naming of the variables is ugly. I will change that.
In this function, yourPRs are passed as Post Form, since it is a http call. In order not to pass the full posts we only pass the significant information (repository_url and number). When we are unmarshalling, we use the same type in order to avoid creating more types.